backend/compact-connect/lambdas/purchases/tests/unit/test_purchase_client.py

```diff
@@ -0,0 +1,396 @@
+            'Compact Privilege for Ohio (Military Discount)',
+            api_contract_v1_obj.transactionRequest.lineItems.lineItem[0].description,
+        )
+
+        # ensure the total amount is the sum of the two line items
+        self.assertEqual(125.50, api_contract_v1_obj.transactionRequest.amount)
+
+    @patch('purchase_client.createTransactionController')
+    def test_purchase_client_raises_failed_transaction_exception_when_transaction_fails(
+        self, mock_create_transaction_controller
+    ):
+        from purchase_client import PurchaseClient
+
+        mock_secrets_manager_client = self._generate_mock_secrets_manager_client()
+        self._when_authorize_dot_net_transaction_fails(
+            mock_create_transaction_controller=mock_create_transaction_controller
+        )
+
+        test_purchase_client = PurchaseClient(secrets_manager_client=mock_secrets_manager_client)
+
+        with self.assertRaises(CCFailedTransactionException):
+            test_purchase_client.process_charge_for_licensee_privileges(
+                order_information=_generate_default_order_information(),
+                compact_configuration=_generate_aslp_compact_configuration(),
+                selected_jurisdictions=_generate_selected_jurisdictions(),
+                user_active_military=False,
+            )
+
+    @patch('purchase_client.createTransactionController')
+    def test_purchase_client_raises_internal_exception_when_api_fails(self, mock_create_transaction_controller):
+        from purchase_client import PurchaseClient
+
+        mock_secrets_manager_client = self._generate_mock_secrets_manager_client()
+        self._when_authorize_dot_net_api_call_fails(
+            mock_create_transaction_controller=mock_create_transaction_controller
+        )
+
+        test_purchase_client = PurchaseClient(secrets_manager_client=mock_secrets_manager_client)
+
+        with self.assertRaises(CCInternalException):
+            test_purchase_client.process_charge_for_licensee_privileges(
+                order_information=_generate_default_order_information(),
+                compact_configuration=_generate_aslp_compact_configuration(),
+                selected_jurisdictions=_generate_selected_jurisdictions(),
+                user_active_military=False,
+            )
+
+    @patch('purchase_client.createTransactionController')
+    def test_purchase_client_voids_transaction_using_authorize_net_processor(self, mock_create_transaction_controller):
+        from purchase_client import PurchaseClient
+
+        mock_secrets_manager_client = self._generate_mock_secrets_manager_client()
+        self._when_authorize_dot_net_transaction_is_successful(
+            mock_create_transaction_controller=mock_create_transaction_controller
+        )
+
+        test_purchase_client = PurchaseClient(secrets_manager_client=mock_secrets_manager_client)
+
+        result = test_purchase_client.void_privilege_purchase_transaction(
```

Comment:
I assume, in real life, this would actually fail since we never created an initial transaction in the first place? (My assumption is that this is completely mocked and not hitting the authorize.net API.)

Reply:
Correct, we do not want to make actual API calls in our local unit testing, so we mock the client as an awkward collaborator. Here we are verifying the data we are sending to the mock client matches expected values, so that if anyone were to make a breaking change in the future this will fail and alert the developer of the breaking change.